src/screens/AppIntroScreen.tsx

```diff
@@ -158,69 +246,60 @@ export const AppIntroScreen = ({ setOnboardingComplete, onlyTermsAndConditions }
   }
 
   return (
-    <SafeAreaViewFlex style={styles.background}>
+    <SafeAreaViewFlex style={[styles.background, { backgroundColor }]}>
       <StatusBar style="dark" translucent backgroundColor="transparent" />
       <AppIntroSlider<AppIntroSlide>
         activeDotStyle={onlyTermsAndConditions ? styles.hiddenDot : styles.activeDot}
+        bottomButton
         data={slides}
         dotClickEnabled={false}
         dotStyle={onlyTermsAndConditions ? styles.hiddenDot : styles.inactiveDot}
         keyExtractor={keyExtractor}
         onDone={setOnboardingComplete}
-        onSlideChange={(_, index) => {
-          // hide the next button on the terms and conditions slide
-          if (
-            index + 1 ==
-            slides.findIndex((slide) => slide.onLeaveSlideName === Initializer.TermsAndConditions)
-          ) {
-            setShowNextButtonTermsAndConditions(false);
-          }
-
-          slides[index]?.onLeaveSlide?.(true);
+        onSlideChange={handleSlideChange}
+        renderDoneButton={() => <SliderButton label={texts.appIntro.continue} isLightest />}
+        renderItem={renderItem}
+        onSkip={handleSkip}
+        ref={slider}
+        renderNextButton={() => {
+          const currentSlideIndex = slider.current?.state?.activeIndex ?? 0;
+          const isTermsSlide =
+            slides[currentSlideIndex]?.onLeaveSlideName === Initializer.TermsAndConditions;
+
+          const buttonStyle =
+            isTermsSlide && !hasAcceptedTerms
+              ? [styles.sliderButtonContainer, styles.sliderButtonDisabled]
+              : styles.sliderButtonContainer;
+
+          return (
+            <TouchableOpacity
+              onPress={() => {
+                if (isTermsSlide && !hasAcceptedTerms) {
+                  termsAndConditionsAlert();
+
+                  return;
+                }
+
+                slider.current?.goToSlide(currentSlideIndex + 1, true);
+              }}
+            >
+              <View style={buttonStyle}>
+                <BoldText lightest={!(isTermsSlide && !hasAcceptedTerms)}>
+                  {texts.appIntro.continue.toUpperCase()}
+                </BoldText>
+              </View>
+            </TouchableOpacity>
+          );
```

Comment:
please make this a component like `SliderButton` to not have so much code here in the render part. it is very hard to read.